cclib/parser/turbomoleparser.py

```diff
@@ -165,6 +165,32 @@ def extract(self, inputfile, line):
 
             # We have entered a new module (sub program); reset our success flag.
             self.metadata['success'] = False
+            
+        # Solvation.
+        if "COSMO switched on" in line:
+            self.metadata['solvent_model'] = "COSMO"
+            
+        elif "COSMO RESULTS" in line:
+            self.metadata['solvent_model'] = "COSMO"
+            line = next(inputfile)
+            line = next(inputfile)
+            
+            while set(line.strip()) != set("="):
+                line = next(inputfile)
+                
+                if "solvent_params" not in self.metadata:
+                    self.metadata['solvent_params'] = {}
+                
+                elif "epsilon:" in line:
+                    self.metadata['solvent_params']['epsilon'] = float(line.split()[-1])
+                    
+                elif "refind:" in line:
+                    self.metadata['solvent_params']['refractive_index'] = float(line.split()[-1])
+                
+                # PLEASE REVIEW: Not sure what this is or what name would be suitable,
+                # guessing it's useful tho...
+                elif "fepsi:" in line:

```

Comment:
I think it's Klamt's function in https://manual.q-chem.com/latest/subsec_COSMO-Klamt.html.  I'm on my phone and can't check.  Try calculating it from the static epsilon and compare.

Reply:
Ah thanks, yeah that looks right. I'll leave the name as `f_epsilon` so it's compatible with other definitions (rather than calling it the COSMO specific Klamt's function)